app/validators/digital_object/enabled_dynamic_fields_validations.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+module DigitalObject::EnabledDynamicFieldsValidations
+  # Returns any errors related to project level validations.
+  #
+  # @return [String] if there are errors
+  # @return false if there are no errors
+  def enabled_field_errors(digital_object, data)
+    errors = []
+    dynamic_field_paths = []
+    data.each do |df_group, children|
+      dynamic_field_paths = collect_field_paths(df_group, children, dynamic_field_paths)
+      dynamic_field_paths.each do |dynamic_field_path|
+        if (e = not_enabled?(dynamic_field_path, digital_object))
+          errors.concat e.map { |i| [dynamic_field_path, i] }
+        end
+      end
+    end
+
+    errors = check_required_fields(dynamic_field_paths, digital_object, errors)
```

Comment:
This line assigns a new value to the `errors` object, but internally the `check_required_fields` method also modifies the passed-in errors object.